lib/swaps/SwapClientManager.ts

```diff
@@ -0,0 +1,224 @@
+import Config from '../Config';
+import { Models } from '../db/DB';
+import BaseClient from '../BaseClient';
+import LndClient from '../lndclient/LndClient';
+import { LndLogger, LndPubKeys, LndInfos } from '../lndclient/types';
+import RaidenClient from '../raidenclient/RaidenClient';
+import { RaidenInfo } from '../raidenclient/types';
+import Logger, { Loggers } from '../Logger';
+import Pool from '../p2p/Pool';
+import { SwapClient } from '../constants/enums';
+import OrderBookRepository from '../orderbook/OrderBookRepository';
+import errors from './errors';
+import { Currency } from '../orderbook/types';
+
+class SwapClientManager {
+  /** A map between currencies and all swap clients */
+  private swapClients = new Map<string, BaseClient>();
+  private raidenClient!: RaidenClient;
+  private repository: OrderBookRepository;
+
+  constructor(
+    private models: Models,
+    private config: Config,
+    private loggers: Loggers,
+    private pool: Pool,
+  ) {
+    this.repository = new OrderBookRepository(loggers.orderbook, models);
+  }
+
+  /**
+   * Wraps each lnd logger call with currency.
+   * @returns A wrapped lnd logger object.
+   */
+  private static wrapLndLogger = (logger: Logger, currency: string): LndLogger => {
+    return {
+      error: (msg: string) => logger.error(`${currency}: ${msg}`),
+      warn: (msg: string) => logger.warn(`${currency}: ${msg}`),
+      info: (msg: string) => logger.info(`${currency}: ${msg}`),
+      verbose: (msg: string) => logger.verbose(`${currency}: ${msg}`),
+      debug: (msg: string) => logger.debug(`${currency}: ${msg}`),
+      trace: (msg: string) => logger.trace(`${currency}: ${msg}`),
+    };
+  }
+
+  /**
+   * Starts all swap clients, binds event listeners
+   * and waits for the swap clients to initialize.
+   * @returns A promise that resolves upon successful initialization, rejects otherwise.
+   */
+  public async init(): Promise<void> {
+    const initPromises = [];
+    // setup LND clients and initialize
+    for (const currency in this.config.lnd) {
+      const lndConfig = this.config.lnd[currency]!;
+      if (!lndConfig.disable) {
+        const lndClient = new LndClient(
+          lndConfig,
+          currency,
+          (SwapClientManager.wrapLndLogger(this.loggers.lnd, currency) as Logger),
+        );
+        this.swapClients.set(currency, lndClient);
+        initPromises.push(lndClient.init());
+      }
+    }
+    // setup raiden client and connect if configured
+    const raidenEnabled = !this.config.raiden.disable;
+    if (raidenEnabled) {
+      this.raidenClient = new RaidenClient(this.config.raiden, this.loggers.raiden);
+      initPromises.push(this.raidenClient.init());
+    }
+    // bind event listeners
+    this.bind();
+    await Promise.all(initPromises);
```

Comment:
It might make sense to remove swap clients that turned out to be disabled after the `initPromises` resolve, since we currently treat disabled clients as if they don't exist I believe.

Reply:
Added a TODO for this since it's not a trivial change. We'd also need to readd the swap clients once their state becomes enabled.